explorer/explorer.go

```diff
@@ -217,9 +223,42 @@ func New(dataSource explorerDataSourceLite, primaryDataSource explorerDataSource
 	return exp
 }
 
-func (exp *explorerUI) Store(blockData *blockdata.BlockData, _ *wire.MsgBlock) error {
+// prePopulateChartsData should run in the background the first time the
+// system is initialized and consecutive times when a new block is added.
+// Its called at intervals of 5 new blocks.
+func (exp *explorerUI) prePopulateChartsData() {
+	log.Info("Pre-populating the charts data. This may take a minute...")
+	var err error
+
+	CacheChartsData, err = exp.explorerSource.GetPgChartsData()
```

Comment:
CacheChartsData has to be protected from concurrent writes.

Reply:
ok. Noted